runtime/cudaq/algorithms/run.h

```diff
@@ -97,15 +98,19 @@ std::vector<RESULT> run(std::size_t shots,
                         ARGS &&...args) {
   if (shots == 0)
     return {};
-
+#if defined(CUDAQ_REMOTE_SIM)
```

Comment:
Shouldn't this depend on the target selected (i.e. dynamic selection) instead of being baked in as a constant at runtime compile-time?

Reply:
Yea, it should. Thanks to @annagrin, I'm now aware of the `get_remote_capabilities().isRemoteSimulator` field :)  
Fixed in https://github.com/NVIDIA/cuda-quantum/pull/2772/commits/9c9f1ba14c08cd2b81d7531690bd8cc90792aac1